apps/api/src/provider/services/jwt-token/jwt-token.service.ts

```diff
@@ -0,0 +1,60 @@
+import { createSignArbitraryAkashWallet, JwtToken } from "@akashnetwork/jwt";
+import type { AccessScope, JwtTokenPayload } from "@akashnetwork/jwt/src/types";
+import { minutesToSeconds } from "date-fns";
+import { singleton } from "tsyringe";
+import * as uuid from "uuid";
+
+import { Wallet } from "@src/billing/lib/wallet/wallet";
+import { BillingConfig, InjectBillingConfig } from "@src/billing/providers";
+import { Memoize } from "@src/caching/helpers";
+
+const JWT_TOKEN_TTL_IN_SECONDS = 30;
+
+type JwtTokenWithAddress = {
+  jwtToken: JwtToken;
+  address: string;
+};
+
+type GenerateJwtTokenParams = {
+  walletId: number;
+  leases: JwtTokenPayload["leases"];
+  ttl?: number;
+};
+
+@singleton()
+export class JwtTokenService {
```

Comment:
I think this name can cause quite some confusion with console-api server auth. Would it make sense to rename it to something like 
```suggestion
export class ProviderJwtTokenService {
```